bundles/org.openhab.binding.teslapowerwall/src/main/resources/OH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,170 @@
+			<channel id="battery-inst-power" typeId="battery-inst-power"/>
+			<channel id="home-inst-power" typeId="home-inst-power"/>
+			<channel id="solar-inst-power" typeId="solar-inst-power"/>
+			<channel id="grid-energy-exported" typeId="grid-energy-exported"/>
+			<channel id="battery-energy-exported" typeId="battery-energy-exported"/>
+			<channel id="home-energy-exported" typeId="home-energy-exported"/>
+			<channel id="solar-energy-exported" typeId="solar-energy-exported"/>
+			<channel id="grid-energy-imported" typeId="grid-energy-imported"/>
+			<channel id="battery-energy-imported" typeId="battery-energy-imported"/>
+			<channel id="home-energy-imported" typeId="home-energy-imported"/>
+			<channel id="solar-energy-imported" typeId="solar-energy-imported"/>
+			<channel id="full-pack-energy" typeId="full-pack-energy"/>
+			<channel id="degradation" typeId="degradation"/>
+		</channels>
+
+		<config-description>
+			<parameter name="hostname" type="text" required="true">
+				<label>Hostname/IP Address</label>
+				<description>The host name or IP address of the Tesla Powerwall.</description>
+			</parameter>
+			<parameter name="email" type="text" required="true">
+				<label>Email</label>
+				<description>Local Powerwall Account Email</description>
+			</parameter>
+			<parameter name="password" type="text" required="true">
+				<label>Password</label>
+				<description>Local Powerwall Account Password</description>
+				<context>password</context>
+			</parameter>
+			<parameter name="refresh" type="integer" required="false">
+				<label>Refresh Interval</label>
+				<description>Specifies the refresh interval in seconds.</description>
+				<default>10</default>
+			</parameter>
+		</config-description>
+
+	</thing-type>
+
+	<channel-type id="grid-status">
+		<item-type>String</item-type>
+		<label>Grid Status</label>
+		<description>Current status of the Power Grid</description>
+		<state readOnly="true"/>
+	</channel-type>
+	<channel-type id="grid-services">
+		<item-type>Switch</item-type>
+		<label>Grid Services</label>
+		<description>Grid Services Activation Status</description>
+		<state readOnly="true"/>
+	</channel-type>
+	<channel-type id="battery-soe">
+		<item-type>Number:Dimensionless</item-type>
+		<label>Battery SOE</label>
+		<description>Current battery state of charge</description>
+		<state readOnly="true" pattern="%.1f %%"/>
+	</channel-type>
+	<channel-type id="mode">
+		<item-type>String</item-type>
+		<label>Operating Mode</label>
+		<description>Current operating mode</description>
+	</channel-type>
+	<channel-type id="reserve">
+		<item-type>Number:Dimensionless</item-type>
+		<label>Battery Reserve</label>
+		<description>Current battery reserve %</description>
+		<state readOnly="true" pattern="%.1f %%"/>
+	</channel-type>
+	<channel-type id="grid-inst-power">
+		<item-type>Number:Power</item-type>
+		<label>Instant Grid Power</label>
+		<description>Instantaneous Grid Power Supply</description>
```

Comment:
Descriptions do not need to be capitalized like labels (applies to others as well):
```suggestion
		<description>Instantaneous grid power supply</description>
```

For consistency this should probably be changed to be aligned with line 82 which is not capitalized either (correctly).

Reply:
Fixed.